spring-integration-http/src/main/java/org/springframework/integration/http/outbound/ReactiveHttpRequestExecutingMessageHandler.java

```diff
@@ -0,0 +1,198 @@
+	 * @param uriExpression A SpEL Expression that can be resolved against the message object and
+	 * {@link BeanFactory}.
+	 * @param webClient The WebClient to use.
+	 */
+	public ReactiveHttpRequestExecutingMessageHandler(Expression uriExpression, WebClient webClient) {
+		super(uriExpression);
+		this.webClient = (webClient == null ? WebClient.create() : webClient);
+		this.setAsync(true);
+	}
+
+	@Override
+	public String getComponentType() {
+		return (isExpectReply() ? "http:outbound-reactive-gateway" : "http:outbound-reactive-channel-adapter");
+	}
+
+	@Override
+	protected Object exchange(URI uri, HttpMethod httpMethod, HttpEntity<?> httpRequest, Object expectedResponseType,
+			Message<?> requestMessage) {
+		WebClient.UriSpec uriSpec;
+
+		// TODO use WebClient.method(HttpMethod) in the future version
+
+		switch (httpMethod) {
+			case GET:
+				uriSpec = this.webClient.get();
+				break;
+			case HEAD:
+				uriSpec = this.webClient.head();
+				break;
+			case POST:
+				uriSpec = this.webClient.post();
+				break;
+			case PUT:
+				uriSpec = this.webClient.put();
+				break;
+			case PATCH:
+				uriSpec = this.webClient.patch();
+				break;
+			case DELETE:
+				uriSpec = this.webClient.delete();
+				break;
+			case OPTIONS:
+				uriSpec = this.webClient.options();
+				break;
+			case TRACE:
+				throw new UnsupportedOperationException("WebClient doesn't support the TRACE HTTP method");
+			default:
+				throw new UnsupportedOperationException("Unsupported HTTP method");
+		}
+
+		WebClient.HeaderSpec spec = uriSpec.uri(uri)
+				.headers(httpRequest.getHeaders());
+
+		Mono<ClientResponse> responseMono;
+		if (httpRequest.hasBody()) {
+			responseMono = spec.exchange(BodyInserters.fromObject(httpRequest.getBody()));
+		}
+		else {
+			responseMono = spec.exchange();
+		}
+
+		if (isExpectReply()) {
+
+			ResolvableType responseType;
+
+			if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
+				responseType = ResolvableType.forType(((ParameterizedTypeReference<?>) expectedResponseType).getType());
+			}
+			else if (expectedResponseType != null) {
+				responseType = ResolvableType.forClass((Class<?>) expectedResponseType);
+			}
+			else {
+				responseType = null;
+			}
+
+			return responseMono
+					.map(response ->
+							new ResponseEntity<>(responseType != null
+									? response.body(BodyExtractors.toMono(responseType)).block()
+									: null,
+									response.headers().asHttpHeaders(),
+									response.statusCode()))
+					.map(this::getReply);
+		}
+		else {
+			responseMono
+					.doOnNext(response -> {
+						HttpStatus httpStatus = response.statusCode();
+						if (httpStatus.is4xxClientError() || httpStatus.is5xxServerError()) {
+							throw new WebClientException(
+									"ClientResponse has erroneous status code: " + httpStatus.value() +
+											" " + httpStatus.getReasonPhrase());
+						}
+					})
+					.subscribe(v -> {}, ex -> sendErrorMessage(requestMessage, ex));
```

Comment:
Uh! There is a couple of Checkstyle violations, too:
```
:spring-integration-http:checkstyleMain[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-http/src/main/java/org/springframework/integration/http/outbound/ReactiveHttpRequestExecutingMessageHandler.java:192:58: '{' is not followed by whitespace. [WhitespaceAround]

[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-http/src/main/java/org/springframework/integration/http/outbound/ReactiveHttpRequestExecutingMessageHandler.java:192:58: '}' is not preceded with whitespace. [WhitespaceAround]
```

This must be like `v -> { }` 😄 

Please, fix this on merge as well, @garyrussell